System/Random.hs

```diff
@@ -970,16 +970,13 @@ instance UniformRange Char where
 instance Random Bool where
   randomM = uniform
 instance Uniform Bool where
-  uniform = uniformR (minBound, maxBound)
+  uniform = fmap wordToBool . uniformWord8
+    where wordToBool w = (w .&. 1) /= 0
 instance UniformRange Bool where
-  uniformR (a, b) g = int2Bool <$> uniformR (bool2Int a, bool2Int b) g
-    where
-      bool2Int :: Bool -> Int
-      bool2Int False = 0
-      bool2Int True  = 1
-      int2Bool :: Int -> Bool
-      int2Bool 0 = False
-      int2Bool _ = True
+  uniformR (False, False) _g = return False
+  uniformR (False, True)   g = uniform g
+  uniformR (True, False)   g = uniform g
+  uniformR (True, True)   _g = return True
```

Comment:
How about:
```haskell
  uniformR (False, False) _g = return False
  uniformR (True, True)   _g = return True
  uniformR _               g = uniform g
```

This is unlikely to improve the performance, but will simplify things to do for the compiler :)

Reply:
Done. I benchmarked it - no difference.